src/app/chat/views/channels_list.nim

```diff
@@ -103,9 +103,16 @@ QtObject:
     self.endResetModel()
 
   proc addChatItemToList*(self: ChannelsList, channel: Chat): int =
-    self.beginInsertRows(newQModelIndex(), 0, 0)
-    self.chats.insert(channel, 0)
-    self.endInsertRows()
+    var found = false
+    for chat in self.chats:
+      if chat.id == channel.id:
+        found = true
```

Comment:
should you quit/break the loop once found?

Reply:
yep, you're right ;) updated